oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtAuthenticationConverter.java

```diff
@@ -16,57 +16,53 @@
 
 package org.springframework.security.oauth2.server.resource.authentication;
 
-import java.util.Arrays;
 import java.util.Collection;
-import java.util.Collections;
-import java.util.stream.Collectors;
 
 import org.springframework.core.convert.converter.Converter;
 import org.springframework.security.authentication.AbstractAuthenticationToken;
 import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.oauth2.jwt.Jwt;
-import org.springframework.util.StringUtils;
+import org.springframework.util.Assert;
 
 /**
  * @author Rob Winch
  * @author Josh Cummings
  * @since 5.1
  */
 public class JwtAuthenticationConverter implements Converter<Jwt, AbstractAuthenticationToken> {
-	private static final String SCOPE_AUTHORITY_PREFIX = "SCOPE_";
-
-	private static final Collection<String> WELL_KNOWN_SCOPE_ATTRIBUTE_NAMES =
-			Arrays.asList("scope", "scp");
-
+	private Converter<Jwt, Collection<GrantedAuthority>> grantedAuthoritiesConverter
+			= new JwtGrantedAuthoritiesConverter();
 
+	@Override
 	public final AbstractAuthenticationToken convert(Jwt jwt) {
 		Collection<GrantedAuthority> authorities = extractAuthorities(jwt);
 		return new JwtAuthenticationToken(jwt, authorities);
 	}
 
+	/**
+	 * Extracts the {@link GrantedAuthority}s from scope attributes typically found in a {@link Jwt}
+	 *
+	 * @param jwt The token
+	 * @return The collection of {@link GrantedAuthority}ies found on the token
```

Comment:
Since I've already asked you to change something in this file :), I'll also ask that we please be consistent on pluralization here. Spring Security isn't really consistent across the board with how GrantedAuthorities are pluralized, but it would be good to be at least consistent inside the same file. Would you please have this one be `{@link GrantedAuthority}s`, like you have in the method description?